acquire-video-runtime/src/runtime/channel.c

```diff
@@ -198,7 +198,7 @@ channel_read_map(struct channel* self, struct channel_reader* reader)
     // here. A call to channel_read_unmap() would return early and not advance
     // the reader's bookmarks in that case, so we need to do it here.
     if (!nbytes) {
-        goto AdvanceToWriterHead;
+        goto AdvanceTailToReaderPos;
```

Comment:
As discussed, replace this w your fix

```c    
    out = 0;
    nbytes = 0;
    *pos = self->head;
    *tail = reader->pos;
```

and use an `else` to leave the `reader->state` Unmapped.